### DISCUSSION TITLE
Modal focus on close

### DISCUSSION THREAD

**Discussion by palmem29** [twbs/bootstrap#41097] [Q&A] [answered]
It appears Bootstrap's modal nicely shifts focus from the closed modal back to the <button> or <a> that triggered it. So I understand focus will be returned easily for this implementation:

```
<button type="button" class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#exampleModal">
  Launch demo modal
</button>
```

But if we're implementing a modal like this?

```
$("#exampleModalTrigger").on("click", function () {
    $("#exampleModal").modal("show");
});
```

Is there a promise made that Bootstrap will get focus back to #exampleModalTrigger? Or does it only work with data attributes? Or will it work if I go the vanilla JS route? Or is it meant to work with all of these tactics and there's something else about my implementation that I need to troubleshoot?

**Comment by julien-deramond** [comment, marked as answer]
When `data-bs-toggle="modal"` is used, the code automatically gives back the focus to the element that triggered the modal. It's done in our code at https://github.com/twbs/bootstrap/blob/a353ed4b19a6849d021aa23b4bd6395f6bf3c2bb/js/src/modal.js#L352-L356

When you don't use `data-bs`, you'll have to do that manually. I've created a [StackBlitz showing how to give back the focus in JS](https://stackblitz.com/edit/kjkpah64?file=index.html,index.js).

The code could be improved, but basically, it can be done like this:

```js
(() => {
  'use strict';

  const btnSecondary = document.getElementById('btnSecondary');
  const exampleModal = document.getElementById('exampleModal');
  const myModal = new bootstrap.Modal(exampleModal, {});

  btnSecondary.addEventListener('click', (event) => {
    if (myModal) {
      exampleModal.addEventListener('hidden.bs.modal', (event) => {
        btnSecondary.focus();
      });
      myModal.show();
    }
  });
})();
```

The idea here is to listen to the `hidden.bs.modal` event.

In jQuery, could be something like this (haven't tested):

```js
$('#exampleModal').on('hidden.bs.modal', function () {
    $('#btnSecondary').focus();
});
```

**Reply by palmem29** [reply]
Excellent! Thank you for your insight!

**Comment by jonhubby** [comment]
Thanks, brother!
